### PR TITLE
Note for BinarySerializable

### LINKED ISSUE #817: Document update for BinarySerializable  attribute.
Update the documentation for BinarySerializable  attribute as BinarySerialization is no longer supported in .NET 8.

(https://learn.microsoft.com/en-us/dotnet/core/compatibility/serialization/8.0/binaryformatter-disabled) due to [security concerns](https://learn.microsoft.com/en-au/dotnet/standard/serialization/binaryformatter-security-guide)

### PR DESCRIPTION
**Pull request by ashishdawale20**
Note for BinarySerializable.

Resolves #817

### CODE REVIEW

**Comment by SeanKilleen** [comment]
@OsirisTerje did we officially decide to remove this from NUnit 4.x? Just making sure because I can't find anything in the NUnit repo discussing it.

File: docs/articles/nunit/writing-tests/constraints/BinarySerializableConstraint.md [hunk 1]
```diff
@@ -23,3 +23,8 @@ Assert.That(someObject, Is.BinarySerializable));
 ## See also
 
 * [XmlSerializableConstraint](XmlSerializableConstraint.md)
+
+## Note
+Binary Serialization is no longer supported in .NET 8 due to security concerns, from Nunit4 this constraint is being removed.
+https://learn.microsoft.com/en-us/dotnet/core/compatibility/serialization/8.0/binaryformatter-disabled
+https://learn.microsoft.com/en-au/dotnet/standard/serialization/binaryformatter-security-guide
```
Comment (SeanKilleen) [inline docs/articles/nunit/writing-tests/constraints/BinarySerializableConstraint.md:30]:
```suggestion
> [!WARNING]
> Binary Serialization is no longer supported in .NET 8 due to security concerns. From NUnit 4.x onward, this constraint is not available. More information can be found in these Microsoft Learn articles [on BinaryFormatter's removal](https://learn.microsoft.com/en-us/dotnet/core/compatibility/serialization/8.0/binaryformatter-disabled) and [on the BinaryFormatter security guide](https://learn.microsoft.com/en-au/dotnet/standard/serialization/binaryformatter-security-guide)

```

Minor tweaks to formatting & language

**Review by SeanKilleen — CHANGES_REQUESTED** [review]
Some minor changes suggested, and also I want to confirm it's being remove.d

**Comment by SeanKilleen** [comment]
@ashishdawale20 no worries on the changes, I already made them.

Thank you for linking the discussion! I thought I'd seen it somewhere but couldn't find it.

I will approve this and get it merged.

**Review by SeanKilleen — APPROVED** [review]
Thanks @ashishdawale20!

**Comment by OsirisTerje** [comment]
@SeanKilleen This issue/comment https://github.com/nunit/nunit/pull/4511#issuecomment-1765896032

**Pull request merged**